examples/nextjs/providers/EventDispatcher.js

```diff
@@ -0,0 +1,51 @@
+import React, { useEffect, useReducer } from 'react';
+
+export const EventLogContext = React.createContext();
+
+const eventReducer = (state, action) => {
+  if (
+    action.type === 'PAGE_VIEW' ||
+    action.type === 'PRODUCT_VIEW' ||
+    action.type === 'ADD_TO_CART' ||
+    action.type === 'REMOVE_FROM_CART' ||
+    action.type === 'CHECKOUT_INIT'
+  ) {
+    return [...state, action];
+  }
+  return state;
+};
+
+export const EventDispatcherProvider = ({ children }) => {
+  const [eventLog, dispatchEvent] = useReducer(eventReducer, []);
+
+  useEffect(() => {
+    if (eventLog.length > 0 && process.browser) {
```

Comment:
`process.browser` is [deprecated](https://github.com/vercel/next.js/issues/5354#issuecomment-520305040) in Next.js - instead, use `typeof window !== 'undefined'`